File: tabulation/how_sum/python_impl/how_sum.py
def how_sum(target, numbers):
    table = [None]*(target+1)
    table[0] = []

    for index, element in enumerate(table):
        if element != None:
            for number in numbers:
                if (index+number) <= target:
                    # a = [ele for ele in table[index]]
                    # a.append(number)
                    a = [*table[index], number]
                    table[index+number] = a

    return table[target]


print(how_sum(7, [2, 3]))
print(how_sum(7, [5, 3, 4, 7]))
print(how_sum(7, [2, 4]))
print(how_sum(8, [2, 3, 5]))
print(how_sum(300, [7, 14]))

File: tabulation/how_sum/js_impl/howSum.js
const howSum = (target, numbers) => {
    const table = Array(target + 1).fill(null);
    table[0] = [];
    for (let i = 0; i < table.length; i++) {
        if (table[i] !== null) {
            for (let element of numbers) {
                if ((i + element) <= target) {
                    table[i + element] = [...table[i], element];
                }

            }
        }
    }
    return table[target];
}


console.log(howSum(7, [2, 3])) //[3,2,2]
console.log(howSum(7, [5, 3, 4, 7])) //[4,3]
console.log(howSum(7, [2, 4])) //null
console.log(howSum(8, [2, 3, 5])) //[2,2,2,2]
console.log(howSum(300, [7, 14])) //null